PlayFabClientSDK/source/ITransportPlugin.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using PlayFab.Internal;
+
+namespace PlayFab
+{
+    /// <summary>
+    /// Interface of any transport SDK plugin.
+    /// </summary>
+    public interface ITransportPlugin: IPlayFabPlugin
+    {
+        Task<object> DoPost(string urlPath, PlayFabRequestCommon request, string authType, string authKey, Dictionary<string, string> extraHeaders);
```

Comment:
I think we should remove authType and authKey parameters here.
That assumes certain meaning about how those auths work, and such.
A more generic HTTPS layer shouldn't have PlayFab auth assumptions.
Likewise, I don't think this should be "extraHeaders" but rather just "headers".

The real question is:
Do we like PlayFabRequestCommon as the "most" generic request?  Or should it just accept "object"?

Reply:
I updated the interface as proposed but started to have mixed feelings about that. Unless we significantly refactor the concept if how we apply authentication headers we are forced to have two different DoPost signatures now - one is more generic and the other one "for PlayFab". 

What if we keep our current signature instead without more generic one? It would mean "interface for a transport that may take authType and authKey parameters". It doesn't have to use them. We still can replace PlayFabRequestCommon with object, though.

---
In reply to: [200494315](https://github.com/PlayFab/CSharpSDK/pull/97#discussion_r200494315) [](ancestors = 200494315)